build/Band_Representations.jl

```diff
@@ -0,0 +1,247 @@
+using LinearAlgebra
+using Crystalline
+using StaticArrays
+
+function group_representative(symop::SymOperation, sg::SpaceGroup)
+    symop_xyzt=xyzt(symop)
+    for i in 1:size(sg)[1]
+        xyztiter=xyzt(sg[i])
+        if xyztiter==symop_xyzt
+            return i
+        end
+    end
+end
+
+function _find_isomorphic_parent_pointgroup(G)
+    D = dim(G)
+    ctᴳ = MultTable(G).table
+    @inbounds for iuclab in PGS_IUCs[D]
+        P = pointgroup(iuclab, D)
+        ctᴾ = MultTable(P).table
+        if ctᴳ == ctᴾ # bit sloppy; would miss ismorphisms that are "concealed" by row/column swaps
+            return P
+        end
+    end
+    return nothing # in case we didn't find any isomorphic parent
+end
+
+function in_site_symmetry(sitegroup::SiteGroup, symop::SymOperation)
+    #index= findfirst(x -> x==rotation(symop), rotation.(sitegroup)) 
+    index= findfirst(x -> x==xyzt(symop), xyzt.(sitegroup)) 
+    if index == nothing
+        return 0 
+    else 
+        return index
+    end
```

Comment:
You can simplify this to:
```suggestion
   return findfirst(≈(symop), sitegroup))
```
It's better just to leave the output as `nothing` if nothing was found (and then test for `nothing` equality later instead of testing for whether equal to `0`). There's no performance impact as far as I'm aware of (Julia has optimized `Union{Nothing, T}` types heavily - and small unions generally - precisely for this type of use-case; it's e.g., how loops are rewritten to goto statements).

Since you can rewrite it in this simple form, there's probably no need to have a full function for it (so maybe just delete this function).